### ISSUE TITLE
/src/github.com/haccer/subjack/fingerprints.json: no such file or directory

### ISSUE THREAD

**Issue by ttmyst** [haccer/subjack#25]
Hello

I am using the newest version of subjack with go1.10.4 on Ubuntu, but have the following error: 
/src/github.com/haccer/subjack/fingerprints.json: no such file or directory
Subjack tries to find fingerprints.json by absolute path, not in directory marked as GOPATH.
Could you please help me fixing this? 
I am sorry for this issue if i'm not right.
Thank you

**Comment by madaratech** [comment]
What's the output of this `echo $GOPATH`?  @ttmyst

**Comment by ttmyst** [comment]
@madaratech output is /home/tati/go
subject binary is in /home/tati/go/bin directory, sources are in /home/tati/go/src/github.com/haccer/subjack

**Comment by madaratech** [comment]
No, as par code it looks with your `$GOPATH` https://github.com/haccer/subjack/blob/99fe2fff937aa6aa3d30679c52320fb874f3eec9/main.go#L15

**Comment by madaratech** [comment]
Anyway you can always feed that file location with `-c` so, use `-c /home/tati/go/src/github.com/haccer/subjack/fingerprints.json` while running subjack.

**Comment by ttmyst** [comment]
@madaratech thank you! it works with -c flag
Stranger things :)

**Comment by jdonsec** [comment]
I'm having the same issue on Ubuntu 18.0.4 

"open /src/github.com/haccer/subjack/fingerprints.json: no such file or directory"

I can confirm file and directory exists on my system.

**Comment by haccer** [comment]
Thanks, I will look into this possibly tomorrow @jdonsec @ttmyst.

**Comment by haccer** [comment]
It’s possible this is happening because the environment variable `GOPATH` is not set.

**Comment by jdonsec** [comment]
@haccer this is what I have at the end of file for ~/ .profile

export GOPATH=$HOME/work
export PATH=$PATH:/usr/local/go/bin:$GOPATH/bin

**Comment by SabunMacTavish** [comment]
I faced the same problem, workaround is to use -c AND -v, at the same time.

**Comment by anonymoustpain** [comment]
The GOPATH variable is not resolving the GOPATH using "os.Getenv("GOPATH")" with the default golang install. For bash, edit ~/.bash_profile

export GOPATH=$HOME/go

**Comment by CyberTheReape** [comment]
same problem... i am using go 1.10... gopath is true..
-----------------------------------------------------------------------
subjack -d twitter.com -c fingerprints.json 
2019/04/18 12:34:02 open : no such file or directory
--------------------------------------------------------------------------

**Comment by vipzen** [comment]
the -c flag workarounded for me. thanks.

**Comment by sabeesh03** [comment]
can't load package: package subjack.go: cannot find package "subjack.go" in any of:
        /usr/lib/go-1.14/src/subjack.go (from $GOROOT)
        /home/indira/go/src/subjack.go (from $GOPATH)

**Comment by neilstuartcraig** [comment]
I have the same issue using subjack on MacOS (10.15.4 Catalina) and Go 1.14.4 via `brew`.
I literally followed the install info and ran `go get github.com/haccer/subjack`, made a domains file in `w.txt` and ran `~/go/bin/subjack -w w.txt -a -ssl`:

```
2020/07/15 17:51:03 open /src/github.com/haccer/subjack/fingerprints.json: no such file or directory
2020/07/15 17:51:03 open /src/github.com/haccer/subjack/fingerprints.json: no such file or directory
```

My `$GOPATH` is empty, seems that on MacOS via `brew` that doesn't get set.

I see that I do have `~/go/src/github.com/haccer/subjack/fingerprints.json` and using the `-c` workaround fixes it. Is the issue just the lack of `$GOPATH`? Maybe worth using a default if that doesn't exist?

**Comment by haccer** [comment]
This has been fixed — fingerprints are now embedded directly into the binary, so no external config file is needed.
